src/checkers/EventInfo.ml

```diff
@@ -63,22 +62,20 @@ struct
            | Some tlist ->
               (* verify types match *)
               let printer tplist =
-                List.fold_left
-                  (fun acc (n, t) ->
+                List.fold_left tplist ~init: "[" ~f:(fun acc (n, t) ->
```

Comment:
Isn't there a String.concat version that does this? Something like

```
  "[" :: (List.fold_right tplist ~init:"]" ~f:(fun (n, t) acc ->
             (Printf.sprintf "(%s : %s); " n (pp_typ t)) :: acc))
  |> String.concat
```

Reply:
`tplist` is of type `(string * typ) list`

Reply:
Oh, sorry, misread your comment

Reply:
I am actually quite annoyed there isn't a `map_to_string_and_then_concat` function on lists. That would be useful.

Reply:
A version using `List.map`:
```
let printer tplist =
  "[" ^
   (List.map tplist ~f:(fun (n, t) -> Printf.sprintf "(%s : %s); " n (pp_typ t))
   |> String.concat ~sep:"") ^
  "]"
```

or using a more efficient string buffer (it's not gonna make a difference here, though):

```
let printer tplist =
  let buf = Buffer.create 1024 in
  Buffer.add_char buf '[';
  List.iter tplist ~f:(fun (n, t) ->
        Printf.sprintf "(%s : %s); " n (pp_typ t)
        |> Buffer.add_string buf);
  Buffer.add_char buf ']';
  Buffer.contents buf

```

Reply:
The string buffer solution is overkill when we have `String.concat`.

I OK with either of the other solutions. My concern was about the use of `^`, but I iniitally misread the documentation for `^`, and thought it was quadratic in the length of the resulting list, but it's only quadratic in the number of strings being concatenated, so it's fine to do it this way.